io-engine/src/rebuild/nexus_rebuild.rs

```diff
@@ -30,18 +37,26 @@ use super::{
 /// that there is no concurrent between the nexus and the rebuild.
 /// This is a frontend interface that communicates with a backend runner which
 /// is the one responsible for the read/writing of the data.
-pub struct NexusRebuildJob(RebuildJob);
+pub struct NexusRebuildJob {
+    job: RebuildJob,
+}
+
+pub struct NexusRebuildJobStarter {
+    job: Option<NexusRebuildJob>,
```

Comment:
Why job is  Option ?

Reply:
Because it's taken once we store it. I've added doc comments for this.